lib/searches.js

```diff
@@ -1,252 +1,343 @@
-        console.log("Crawling level 0...");
+    // Written by Nigel, declare result object
+    let result = {
+        status: PENDING,
+        keywordURL: ""
+    };
+    
+    return searchingPromise = new Promise((resolve, reject) => {
+        // Search random links until the searchDepth is reached.
+        function getRandomLink(links, level) {
 
-        // Get random links until the searchDepth is reached.
-        function getRandomLink(url) {
-            
-            // If searchDepth is reached, insert the tree into the database
-            if (currentDepth >= searchDepth) {
-                resolve(1);
+            // If searchDepth is reached, return
+            if (level > searchDepth) {
+                result.status = SUCCESS;
+                resolve(result);
                 return;
-            } 
-            
-            // Call dfsHelper to insert the next random link into the edge array
-            let helperPromise = dfsHelper(nextUrl, edges, currentDepth);
+            }
+
+            // Call getDFSLinks to get links from a valid URL in the current link list
+            let helperPromise = getDFSLinks(links, level, searchDepth, searchID, 0, result, keyword);
 
-            helperPromise.then(() => {
-                // If link error, return results thus far.
-                if (edges[currentDepth] === undefined){
-                    resolve(-1);
+            return helperPromise.then((links) => {
+                // If term is found, return
+                if (result.status == TERM_FOUND) {
+                    resolve(result);
                     return;
+                }  
+                
+                // If link has child links, keep searching 
+                if (links && links.length > 0) {
+                    return getRandomLink(links, level + 1);
                 }
-               
-                if (edges[currentDepth].termFound === true){
-                    resolve(2);
+                
+                // If search hits dead end, return results
+                else{
+                    result.status = INVALID;
+                    resolve(result);
                     return;
                 }
-                // Insert link into temp edges table
-                let tempInsert = dbAPI.insertIntoTempEdges(edges[currentDepth].SourceUrl, edges[currentDepth].DestinationUrl, cookie, 'Ian Dalrymple', 'DFS', searchDepth, rootUrl);   
-            
-                tempInsert.catch((err) =>  {
-                    throw err;
-                });
-               
-                nextUrl = edges[currentDepth].DestinationUrl;
-                currentDepth++;
-                console.log("Crawling level " + currentDepth + "...");
-                
-                // Get another random link recursively.
-                getRandomLink(nextUrl);
-            });
-
-            helperPromise.catch((err) => {
+            }).catch((err) => {
                 reject(err);
```

Comment:
For consistency with BFS, instead of rejecting we should resolve with an invalid result, and then not store invalid results in the database.
```js
catch((err) => {
    console.error(error);
    result.status = INVALID;
    return result;
})
```